analyzers/src/SonarAnalyzer.Common/Helpers/XmlHelper.cs

```diff
@@ -42,5 +45,20 @@ public static XDocument ParseXDocument(string text)
                 && attribute.Value.Equals(value.ToString(), StringComparison.OrdinalIgnoreCase)
                 ? attribute
                 : null;
+
+        public static Location CreateLocation(this XAttribute attribute, string path)
+        {
+            // IXmlLineInfo is 1-based, whereas Roslyn is zero-based
+            var startPos = (IXmlLineInfo)attribute;
+            if (startPos.HasLineInfo())
+            {
+                // LoadOptions.PreserveWhitespace doesn't preserve whitespace inside nodes and attributes => there's no easy way to find full length of a XAttribute.
+                var length = attribute.Name.ToString().Length;
+                var start = new LinePosition(startPos.LineNumber - 1, startPos.LinePosition - 1);
+                var end = new LinePosition(startPos.LineNumber - 1, startPos.LinePosition - 1 + length);
+                return Location.Create(path, new TextSpan(start.Line, length), new LinePositionSpan(start, end));
+            }
+            return null;
```

Comment:
This flow is not covered by tests. You can cover it by calling the name with an attribute created by hand. e.g. `new XAttribute(XName.Get("name"), "A")` does not have line info.